### PR TITLE
Vendoring update for go-discover.

### LINKED ISSUE #4408: Updates to go-discover
We need to vendor in the latest go-discover and get the new packet provider, vsphere provider, and some other misc fixes.

- https://github.com/hashicorp/go-discover/pull/52
- https://github.com/hashicorp/go-discover/pull/43

### PR DESCRIPTION
**Pull request by MagnumOpus21**
Changes made in this PR include:
- Added two new providers : vSphere and packer.
- Fixed misc. changes made to go-discover.

Fixes #4408 .

### CODE REVIEW

**Comment by banks** [comment]
@MagnumOpus21 I assume the vendor update is legit although it's a ton of new code - I guess becuase new transitive deps on large vmware SDK or something.

Can you add the command you used to update this to the PR please? It's always worth sanity checking as a few PRs recently have pulled in way too much stuff with a slight vendor command slip up.

Also, are the new vendors added actually usable in Consul after this PR? If so, can we update the docs too please?

**Comment by MagnumOpus21** [comment]
```
govendor fetch github.com/hashicorp/go-discover/provider/packet
govendor fetch github.com/hashicorp/go-discover/provider/vsphere 

govendor update github.com/hashicorp/go-discover
```
The first two commands were to add in the _vSphere_ and the _packet providers_. The one after the newline was for updating the go-discover package. I was looking at _kardianos/govendor_  looking at how to use them.

**Comment by MagnumOpus21** [comment]
```
github.com/Sirupsen/
        github.com/hashicorp/go-discover/provider/packet/
        github.com/hashicorp/go-discover/provider/vsphere/
        github.com/hashicorp/go-discover/terraform.tfstate
        github.com/hashicorp/go-discover/terraform.tfstate.backup
        github.com/packethost/
        github.com/vmware/
        golang.org/x/crypto/ssh/terminal/
```
This was the git status after the addition of the providers.

**Comment by MagnumOpus21** [comment]
@banks Regarding the documentation, doesn't go-discover's README.md contain instructions on how to use the providers? If I'm not in the right path, what documentation where you alluding to sir.

**Comment by MagnumOpus21** [comment]
I can add the docs for them. ^_^

**Comment by MagnumOpus21** [comment]
Updated the `auto-join` documentation. 👍

**Comment by MagnumOpus21** [comment]
This needs the latest ``` golang.org/x/sys/unix``` or else `make` will fail to produce any binary.

**Comment by pearkes** [comment]
It looks like some Terraform tfstate files snuck in here:

```
vendor/github.com/hashicorp/go-discover/terraform.tfstate
```

And I don't see those in the go-discover repo so guessing they're on your local filesystem.

**Comment by MagnumOpus21** [comment]
Remember the go-discover tests from earlier. They snuck in here. Let me remove them. LUL.

File: vendor/vendor.json [hunk 1]
```diff
@@ -240,8 +262,8 @@
 		{"path":"golang.org/x/net/proxy","checksumSHA1":"QEm/dePZ0lOnyOs+m22KjXfJ/IU=","revision":"02ac38e2528ff4adea90f184d71a3faa04b4b1b0","revisionTime":"2017-07-18T17:12:47Z"},
 		{"path":"golang.org/x/net/trace","checksumSHA1":"u/r66lwYfgg682u5hZG7/E7+VCY=","revision":"d866cfc389cec985d6fda2859936a575a55a3ab6","revisionTime":"2017-12-11T20:45:21Z"},
 		{"path":"golang.org/x/sys/cpu","checksumSHA1":"REkmyB368pIiip76LiqMLspgCRk=","revision":"ad87a3a340fa7f3bed189293fbfa7a9b7e021ae1","revisionTime":"2018-06-18T16:37:21Z"},
-		{"path":"golang.org/x/sys/unix","checksumSHA1":"vlicYp+fe4ECQ+5QqpAk36VRA3s=","revision":"cd2c276457edda6df7fb04895d3fd6a6add42926","revisionTime":"2017-07-17T10:05:24Z"},
-		{"path":"golang.org/x/sys/windows","checksumSHA1":"Pki0iA65Dtsjh4gtabPkp/lOa2I=","revision":"cd2c276457edda6df7fb04895d3fd6a6add42926","revisionTime":"2017-07-17T10:05:24Z"},
+		{"path":"golang.org/x/sys/unix","checksumSHA1":"su2QDjUzrUO0JnOH9m0cNg0QqsM=","revision":"ac767d655b305d4e9612f5f6e33120b9176c4ad4","revisionTime":"2018-07-08T03:57:06Z"},
```
Comment (pearkes) [inline vendor/vendor.json]: Is there any concern upgrading this dependency? If I understand correctly it has strict backwards compatibility promises but wanted to raise it.

Comment (pearkes) [reply vendor/vendor.json]: Noting the origin here: https://github.com/hashicorp/go-discover/blob/master/provider/vsphere/vendor/vendor.json#L26-L27

**Pull request merged**